### ISSUE TITLE
Vorbis image metadata is not read

### ISSUE THREAD

**Issue by RomanHargrave** [ampache/ampache#1599]
# Environment

 * Ampache version: 3e99df1

# Settings

Not relevant. 

# Description

In `gather_media_tags`, Ampache does not search for Vorbis metadata images. Seeing as nearly every album in my collection is in some format that uses Vorbis comments/metadata (FLAC, Opus, Theora), my entire collection is without images.

Some information that I have consulted in the past WRT Vorbis comments is https://xiph.org/vorbis/doc/v-comment.html.

**Comment by wagnered** [comment]
@RomanHargrave 

Almost all (1000's)  of my music files are .flac and have no problem.with fetching and displaying the images from the file's metadata:  I used Easytag to insert all the metadata/images into my flac files.

The switches in config/ampache.cfg.php are relevant for gathering art: "art_order" is one.  also, album_art_* affects the width or height of images that will be added to the db.

What does the log say when attempting to gather art on the folder?

**Comment by RomanHargrave** [comment]
```
2017-09-08 13:27:44 [roman] (Art) -> Searching using:["tags","db","folder","musicbrainz"] 
2017-09-08 13:27:44 [roman] (Art) -> Method used: gather_tags 
2017-09-08 13:27:46 [roman] (Art) -> results: 
2017-09-08 13:27:46 [roman] (Art) -> Method used: gather_db 
2017-09-08 13:27:46 [roman] (Art) -> results: 
2017-09-08 13:27:46 [roman] (Art) -> Method used: gather_folder 
2017-09-08 13:27:46 [roman] (folder_art) -> Opening /srv/music/flac/Vulfpeck/2011 - Mit Peck/Disc 01 and checking for Album Art 
2017-09-08 13:27:46 [roman] (Art) -> results: 
2017-09-08 13:27:46 [roman] (Art) -> Method used: gather_musicbrainz 
2017-09-08 13:27:46 [roman] (Art) -> results: 
2017-09-08 13:27:46 [roman] (Core) -> Cannot create image from empty data 
2017-09-08 13:27:46 [roman] (Core) -> Cannot create image from empty data 
2017-09-08 13:27:47 [roman] (Core) -> Cannot create image from empty data 
2017-09-08 13:27:47 [roman] (Core) -> Cannot create image from empty data 
2017-09-08 13:27:47 [roman] (Core) -> Cannot create image from empty data 
2017-09-08 13:27:47 [roman] (Core) -> Cannot create image from empty data 
2017-09-08 13:27:47 [roman] (Core) -> Cannot create image from empty data 
```

Files in `/srv/music/flac/Vulfpeck/2011 - Mit Peck/Disc 01` have image metadata according to `metaflac`:

```
METADATA block #2
  type: 6 (PICTURE)
  is last: false
  length: 572617
  type: 3 (Cover (front))
  MIME type: image/png
  description: 
  width: 0
  height: 0
  depth: 0
  colors: 0 (unindexed)
  data length: 572576
  data: ...
```

Ignore the width and height fields. They are unused in the official spec, iirc.

**Comment by wagnered** [comment]
>Ignore the width and height fields. They are unused in the official
spec, iirc.

I don't understand why.   I just ran metaflac on one of my own files and
the output indicated both height and width:

METADATA block #3
  type: 6 (PICTURE)
  is last: false
  length: 471465
  type: 3 (Cover (front))
  MIME type: image/jpeg
  description: Badfinger - Wish You Were Here - Front.jpg
  width: 1429
  height: 1417
  depth: 0
  colors: 0 (unindexed)
  data length: 471381
  data:
    00000000: FF D8 FF E0...


I cannot recreate the issue.

>|Cannot create image from empty data|


That looks suspicious: You might try loading the flac file into Easytag
and see if the image is visible.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by RomanHargrave** [comment]
Yeah, they're visible in EasyTag.

The spec does include the W/H fields, but since the embedded image can be any format, the image format ends up containing the dimension info. Some encoders/taggers fill the width/height fields, some don't.

**Comment by RomanHargrave** [comment]
@wagnered what commit hash of ampache are you at?

**Comment by wagnered** [comment]
3e99df.  As I am one of the developers, I must keep my local clone current with the github ampache repository.

**Comment by RomanHargrave** [comment]
Huh. I had a look at getID3 and it seems like it should work too. I'm going to look in to it more.

**Comment by wagnered** [comment]
@RomanHargrave 

 `lib/vendor/james-heinrich/getid3/demos/demo.browse.php` is a demo that allows to select and examine most music files and view its  contents.  Line 16 starting with die... must be commented out.
This exercises the ID3 library and displays all metadata including images.

If you can see  the image with this demo, then the problem probably is in the settings. 

I used `php -S 127.0.0.1:8000` as the server to access the demo file because my main server won't allow access to those folders.

**Comment by RomanHargrave** [comment]
The demo does indeed find images.

What might the settings issue be then, do you suppose?

**Comment by RomanHargrave** [comment]
So ampache _did_ enumerate the embedded art, _but_ it did not format it correctly, or is not serving it correctly.

There are entries on the find_art page for the embedded art, but they are all marked invalid.

**Comment by wagnered** [comment]
The switches I referred to earlier should be it.  I can't remember if
Ampache checks `max_upload_size` in the config file


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by RomanHargrave** [comment]
max_upload_size is currently set to 2MB, which should be more than enough for the 600x600 PNG files.

Here's why the entries are marked invalid:

When ampache collects album art from tags, they are collected as `raw` entries, and `find_art` strips out `raw` entries. Thus, `templates/show_arts.inc.php` gets a whole bunch of empty images.

That being said, removing the aforementioned striping logic still does not work properly. Nor should it fix the fact that ampache isn't automatically using embedded art, since that logic is somewhere else.

**Comment by wagnered** [comment]
If you  are seeing the image on the Artist/Albums page there should be
no need to "find art". Ampache should have already grabbed the image
from the file, during catalog creation, and stored it in the db for
later display. Same goes for the image on the album/songlist page.


73 From  AA1AD
Ernie D
Federal Way, WA | CN87ug

**Comment by RomanHargrave** [comment]
It may have, but it certainly isn't displaying it for the majority of albums. The large majority of my albums have empty (e.g. using placeholder cover) images.

**Comment by wagnered** [comment]
@RomanHargrave,

Sorry to take so long to get back to you.

Just for yucks, I converted one of my mp3 files to ogg and imported it into a catalog. The image was added with no problem and shows up as album art. I used Audacity to convert, but it doesn't transfer the image, so I used Easytag to embed the album art from another song in the album.

You might use an IDE to step through the code:  I use  Eclipse-PDT  to debug my code and can offer help in setting it up..

Sorry I can't be of more help...
